src/meson.build

```diff
@@ -264,146 +264,46 @@ if not(get_option('build_static'))
 	endif
 endif
 
-libbrotli = []
-if get_option('with_brotli')
-	libbrotli = [ compiler.find_library('brotlienc') ]
-	if compiler.has_function('BrotliEncoderCreateInstance', args: defs, dependencies: libbrotli, prefix: '#include <brotli/encode.h>')
-		conf_data.set('HAVE_BROTLI_ENCODE_H', true)
-		conf_data.set('HAVE_BROTLI', true)
-	else
-		error('Couldn\'t find brotli header / library')
-	endif
-endif
+libbrotli = dependency('libbrotlienc', required: get_option('with_brotli'))
+conf_data.set('HAVE_BROTLI_ENCODE_H', libbrotli.found())
+conf_data.set('HAVE_BROTLI', libbrotli.found())
 
-libbz2 = []
-if get_option('with_bzip')
-	libbz2 = [ compiler.find_library('bz2') ]
-	if compiler.has_function('BZ2_bzCompress', args: defs, dependencies: libbz2, prefix: '#include <bzlib.h>')
-		conf_data.set('HAVE_BZLIB_H', true)
-		conf_data.set('HAVE_LIBBZ2', true)
-	else
-		error('Couldn\'t find bz2 header / library')
-	endif
-endif
+libbz2 = dependency('bzip2', required: get_option('with_bzip'))
+conf_data.set('HAVE_BZLIB_H', libbz2.found())
+conf_data.set('HAVE_LIBBZ2', libbz2.found())
 
-if get_option('with_dbi')
-	libdbi = dependency('dbi', required: false)
-	if libdbi.found()
-		libdbi = [ libdbi ]
-	else
-		libdbi = [ compiler.find_library('dbi') ]
-		if not(compiler.has_function('dbi_conn_connect', args: defs, dependencies: libdbi, prefix: '#include <dbi/dbi.h>'))
-			error('Couldn\'t find dbi/dbi.h or dbi_conn_connect in lib dbi')
-		endif
-	endif
-	conf_data.set('HAVE_DBI', true)
-endif
+libdbi = dependency('dbi', required: get_option('with_dbi'))
+conf_data.set('HAVE_DBI', libdbi.found())
 
 libfam = []
-if get_option('with_fam') and not(conf_data.get('HAVE_SYS_INOTIFY_H'))
-	libfam = [ compiler.find_library('fam') ]
-	if not(compiler.has_function('FAMOpen2', args: defs, dependencies: libfam, prefix: '#include <fam.h>'))
-		error('Couldn\'t find fam.h or FAMOpen2 in lib fam')
-	endif
-	conf_data.set('HAVE_FAM_H', true)
+if not conf_data.get('HAVE_SYS_INOTIFY_H')
+	libfam = [ compiler.find_library('fam', required: get_option('with_fam')) ]
+	conf_data.set('HAVE_FAM_H', libfam.found())
```

Comment:
Here and later on, you do two unrelated changes:
- use feature options for simplifying found dependency/library objects
- drop the checks for header presence

You should not drop header checks, I think -- because, it's possible to have a library installed on distros with split runtime/devel packages. The runtime library will be detected by find_library, even if the development headers are not installed, and then the build fails.

Reply:
In the case of dependency, pkgconfig files are typically paired with devel packages.

Reply:
hmm this might not be correct. Removing libev-devel locally but keeping libev throws this error:

```
src/meson.build:310:0: ERROR: C shared or static library 'ev' not found
```

with

```
meson --auto-features=enabled
```

Reply:
Yeah, dependency() is explicitly immune to this problem.

As for removing devel packages, it really depends. Debian ships the .so symlink in the devel package, but other distros with split development packages don't.

This means that find_library() will fail when headers aren't installed... but only on Debian-like systems.

Reply:
I'll test on Debian and see what's up.

Reply:
I can't reproduce on Debian 11.

If I have libpam-dev, find_library succeeds. If I remove it, find_library fails. It's like that with the others as well.